### ISSUE TITLE
NPE: Parameter specified as non-null is null: method mozilla.components.feature.customtabs.AbstractCustomTabsService.mayLaunchUrl, parameter url

### ISSUE THREAD

**Issue by ulfandersson** [mozilla-mobile/android-components#9839]
We are using CustomTabs in our app and experience a crash when firefox is the default browser.
```
Fatal Exception: java.lang.NullPointerException: Parameter specified as non-null is null: method mozilla.components.feature.customtabs.AbstractCustomTabsService.mayLaunchUrl, parameter url
       at android.os.Parcel.readException(Parcel.java:1718)
       at android.os.Parcel.readException(Parcel.java:1662)
       at android.support.customtabs.ICustomTabsService$Stub$Proxy.mayLaunchUrl(ICustomTabsService.java:434)
       at androidx.browser.customtabs.CustomTabsSession.mayLaunchUrl(CustomTabsSession.java:107)
```

The documentation for CustomTabsSession clearly states that url can be null:
https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsSession#mayLaunchUrl(android.net.Uri,%20android.os.Bundle,%20java.util.List%3Candroid.os.Bundle%3E)

The documentation for the abstract CustomTabsService implemented in this project isn't as clear: https://developer.android.com/reference/androidx/browser/customtabs/CustomTabsService#mayLaunchUrl(androidx.browser.customtabs.CustomTabsSessionToken,%20android.net.Uri,%20android.os.Bundle,%20java.util.List%3Candroid.os.Bundle%3E)
However url is marked as Nullable:
```
protected abstract boolean mayLaunchUrl(@NonNull CustomTabsSessionToken sessionToken,
            @Nullable Uri url, @Nullable Bundle extras, @Nullable List<Bundle> otherLikelyBundles);
```

So in conclusion url should be nullable here:
https://github.com/mozilla-mobile/android-components/blob/aee1d759580626088ed89c1dad6b75e3b1d213c9/components/feature/customtabs/src/main/java/mozilla/components/feature/customtabs/AbstractCustomTabsService.kt#L86

**Comment by mcarare** [comment]
This will require upgrading to` androidx.browser:browser:1.3.0`.
